text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+## Alternatives
+
+### Use a `<template>` tag instead of `lwc.dynamic`.
+
+Reference to [POC](https://github.com/salesforce/lwc/pull/3218)
+
+Decided against this approach because the LWC team has placed restrictions on being able to add attributes to `<template>` as it does not have any semantic meaning if no LWC directives are associated. In the past the LWC team has seen component authors attempt to write components wrapped in `<template>` believing it would be rendered to the DOM.
+
+For example the following will not render in the DOM:
+
+```html
+<template>
+    <template class="slds-border">
+        <span>hello world</span>
+    </template>
+</template>
+```
+
+As a result the team has placed restrictions on non-root `<template>` tags to discourage component authors from confusing it as an HTML element that will be rendered in the DOM. Because dynamic components should allow attributes and event listeners to be assigned declaratively this is may further confuse component authors about how to use `<template>` element.
+
+### LightningDynamicElement
+
+Full details can be found [here](https://github.com/salesforce/lwc/pull/3204)
+
+#### Brief summary
+
+`LightningDynamicElement` is a special custom element provided by LWC that serves as a wrapper component around dynamic LWC custom elements. `LightningDynamicElement` tackles the issue of reusing the same custom element with multiple definitions by serving as a wrapper around dynamic components that renders their content to the DOM.
+
+`LightningDynamicElement` works by synthetically handling the lifecycle of the dynamic component, which means the dynamic component does not go through the same mounting and unmounting process as other custom elements.  Additionally, `LightningDyanmicElement` does not unmount when the dynamic constructor changes.
+
+_This is similar to the [Angular approach to dynamic components](https://angular.io/guide/dynamic-component-loader)._
+
+#### Comparison with `<lwc.dynamic lwc:is={expression}>`
+
+With `LightningDynamicElement` the framework keeps track of references to the dynamic components and also manages its lifecycle.  In contrast with `<lwc.dynamic>` the component author is responsible for keeping track of the custom element.  This means that any dynamic component must be retrieved each time the component changes.  Additionally, any imperatively assigned attributes or event listeners will need to be reapplied once the component changes as well.
+
+## Adoption strategy
+
+This is a breaking change for existing user of `lwc:dynamic` as it supersedes the directive.  However, `lwc:dynamic` directive has not reached GA and all consumers of the directive are internal.  They will need to switch their components to change the tag names to `lwc.dynamic` and the directives from `lwc:dynamic` to `lwc:is`.
```

Comment:
Let's call out that we will support both `lwc:dynamic` and `lwc:is` directives at the same time to help with the transition. We will also need to add a compilation warning when the compiler detects the usage of the `lwc:dynamic` directive.

Reply:
Added a [section](https://github.com/salesforce/lwc-rfcs/pull/71/files#diff-a1de05148a61ea9b00f70bd4ff5feb3895c16dd99707b173ba57c081b859250bR325) to indicate we will support both directives.